.github/scripts/produce-report.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import os
+import gzip
+import json
+import sys
+from collections import defaultdict
+
+USER_QUOTA = 8_000_000_000
+
+# TODO trash files
+
+def generate_statistics(input_file):
+    # Load the JSON data from the compressed file
+    with gzip.open(input_file, 'rt', encoding='utf-8') as gz_file:
+        data = json.load(gz_file)
+
+    # Dictionary to hold statistics per leading directory
+    stats = {
+        "directories": defaultdict(lambda: {"total_size": 0, "file_count": 0}),
+        "total": {"total_size": 0, "file_count": 0},
+    }
+
+
+    # Process each file's metadata
+    for file_metadata in data["files"]:
+        # Get the leading directory (first part of the relative path)
+        leading_dir = file_metadata["path"].split(os.sep)[0]
```

Comment:
here, since listing of files is in a logical progression, in one pass you could compute sizes for all folders, not "leading" top level only.  So take the folder name "full" for the file. If *it is different from previous* file you considered it could be two things (below all letters are directories):

- you went **deeper** (e.g. prior one was `a/b` and now it is `a/b/c/d`). You just add file sizes to 'a/b/c/d' directory statistic.
- you stayed at the same level but different (e.g. prior `a/b` and now `a/c`) -- this is the same as below -- the common root is `a` so you just do that logic from `a` and go "deeper" into `c` (i.e. nothing special)
- you went **higher** (e.g. prior one was `a/b/c/d` and now it is `a`). At that point you need to add the size of all immediate children between common root (a) and prior one, to corresponding subfolder records... in this case -- add a/b/c/d size to a/b/c, then add size of a/b/c to a/b, and add size of `a/b` to a.

generalization, 
- define `prior_common_root` as the `.`
- for next file, 
  - take common path between `prior_common_root` and `current_folder`.  If it is different from `prior_common_root`:
     - if new `common_root` points deeper -- nothing todo to propagate
     - if new `common_root` points higher -- do additions as described above for every subfolder between new common path and `prior_common_root`
- record `prior_common_root = current_folder`
- `stats[current_folder] += file.size ...`